pyogrio/geopandas.py

```diff
@@ -275,7 +275,11 @@ def read_dataframe(
             # Index not asked, no geometry column and no attribute columns: return empty
             return pd.DataFrame()
         elif geometry_name in df.columns:
-            df["geometry"] = from_wkb(df.pop(geometry_name), crs=meta["crs"])
+            wkb_values = df.pop(geometry_name)
+            if PANDAS_GE_20 and wkb_values.dtype != object:
+                # for example ArrowDtype will otherwise create numpy array with pd.NA
+                wkb_values = wkb_values.to_numpy(na_value=None)
```

Comment:
I decided to this only conditionally, because otherwise it would impact performance for the default case (where the binary pyarrow array was already converted to an object dtype column with None values. When specifying an `na_value` in that case, it would check for missing values and overwrite them with None, which be an unnecessary operation).

Checking for "not object dtype" instead of explicitly checking for "is ArrowDtype" felt the most future proof (eg when pandas adds a proper binary dtype (using Arrow under the hood), it might have the same issue but without being an ArrowDtype instance)